packages/gatsby-dev-cli/src/__tests__/watch.js

```diff
@@ -142,11 +172,487 @@ describe(`watching`, () => {
+  `gatsby-transformer-csv`,
+  `gatsby-transformer-documentationjs`,
+  `gatsby-transformer-excel`,
+  `gatsby-transformer-hjson`,
+  `gatsby-transformer-javascript-frontmatter`,
+  `gatsby-transformer-javascript-static-exports`,
+  `gatsby-transformer-json`,
+  `gatsby-transformer-pdf`,
+  `gatsby-transformer-react-docgen`,
+  `gatsby-transformer-remark`,
+  `gatsby-transformer-screenshot`,
+  `gatsby-transformer-sharp`,
+  `gatsby-transformer-sqip`,
+  `gatsby-transformer-toml`,
+  `gatsby-transformer-xml`,
+  `gatsby-transformer-yaml`,
+  `graphql-skip-limit`,
+]
+
+const mockDepsChanges = packagesWithChangedDeps => ({ packageName }) =>
+  Promise.resolve({
+    didDepsChanged: packagesWithChangedDeps.includes(packageName),
+  })
+
+jest.mock(`../utils/check-deps-changes`, () => {
+  return {
+    checkDepsChanges: jest.fn(),
+  }
+})
+
+jest.mock(`../local-npm-registry/publish-package`, () => {
+  return {
+    publishPackage: jest.fn(),
+  }
+})
+
+jest.mock(`../local-npm-registry/install-packages`, () => {
+  return {
+    installPackages: jest.fn(),
+  }
+})
+
+jest.mock(`../utils/promisified-spawn`, () => {
+  return {
+    promisifiedSpawn: jest.fn(() => Promise.resolve()),
+  }
+})
+
+describe(`dependency changs`, () => {
+  const { publishPackage } = require(`../local-npm-registry/publish-package`)
+  const { installPackages } = require(`../local-npm-registry/install-packages`)
+  const { checkDepsChanges } = require(`../utils/check-deps-changes`)
+  const { promisifiedSpawn } = require(`../utils/promisified-spawn`)
+
+  const assertPublish = ({ include = [], exclude = [] }) => {
+    include.forEach(includedPackage => {
+      expect(publishPackage).toBeCalledWith(
+        expect.objectContaining({
+          packageName: includedPackage,
+        })
+      )
+    })
+
+    exclude.forEach(excludedPackage => {
+      expect(publishPackage).not.toBeCalledWith(
+        expect.objectContaining({
+          packageName: excludedPackage,
+        })
+      )
+    })
+  }
+
+  const assertInstall = ({ include = [], exclude = [] }) => {
+    include.forEach(includedPackage => {
+      expect(installPackages).toBeCalledWith(
+        expect.objectContaining({
+          packagesToInstall: expect.arrayContaining([includedPackage]),
+        })
+      )
+    })
+
+    exclude.forEach(excludedPackage => {
+      expect(installPackages).not.toBeCalledWith(
+        expect.objectContaining({
+          packagesToInstall: expect.arrayContaining([excludedPackage]),
+        })
+      )
+    })
+  }
+
+  let realProcess
+  beforeAll(() => {
+    realProcess = global.process
+
+    global.process = {
+      ...realProcess,
+      exit: jest.fn(),
+    }
+  })
+
+  beforeEach(() => {
+    publishPackage.mockClear()
+    installPackages.mockClear()
+    checkDepsChanges.mockClear()
+    promisifiedSpawn.mockClear()
+  })
+
+  afterAll(() => {
+    global.process = realProcess
+  })
+
+  describe(`publishing and installing packages`, () => {
```

Comment:
should we also check if gatsby copy is working?

Reply:
Like checking if `fs.copy` was called from monorepo to local project? If so I think we can do that in `installs from npm before copying files` test (last test from this file) - it also sounds like good idea

Reply:
yes check if copy has been for the correct packages.

so for example we should test
```
checkDepsChanges.mockImplementationOnce(mockDepsChanges([`gatsby`]))
watch(process.cwd(), [`gatsby`, `gatsby-plugin-sharp`], {
  scanOnce: true,
  quiet: true,
  monoRepoPackages,
  localPackages: [`gatsby`, `gatsby-plugin-sharp`],
})
```

now it should install gastby and copy gatsby & gatsby-plugin-sharp

Reply:
gotcha! makes total sense, will add

Reply:
added https://github.com/gatsbyjs/gatsby/pull/14347/commits/36542cfa0b20b719e58f99ca793f0545985b2ced - let me know if this is what you had in mind